src/test/java/com/list/todo/services/UserServiceTest.java

```diff
@@ -0,0 +1,242 @@
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private ShareRepository shareRepository;
+
+    @Mock
+    private TodoListRepository todoListRepository;
+
+    @Mock
+    private FollowerRepository followerRepository;
+
+    @Mock
+    private Pageable pageable;
+
+    @Mock
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @InjectMocks
+    private UserService userService;
+
+    private static final Long currentUserId = 1L;
+
+    @Test
+    public void getUserById_OnExistentUser_ReturnsOptionalOfUser() {
+        //arrange
+        User user = createUser(1);
+        when(userRepository.findById(currentUserId)).thenReturn(Optional.of(user));
+
+        //act
+        Optional<User> returnedUser = userService.getUserById(currentUserId);
+
+        //assert
+        verify(userRepository).findById(currentUserId);
+        Assert.assertEquals(user, returnedUser.get());
+    }
+
+    @Test
+    public void getUserByUsername_OnExistentUser_ReturnsOptionalOfUser() {
+        //arrange
+        User user = createUser(1);
+        when(userRepository.findByUsername(user.getUsername())).thenReturn(Optional.of(user));
+
+        //act
+        Optional<User> returnedUser = userService.getUserByUsername(user.getUsername());
+
+        //assert
+        verify(userRepository).findByUsername(user.getUsername());
+        Assert.assertEquals(user, returnedUser.get());
+    }
+
+    @Test
+    public void getUserInfo_OnExistentUser_ReturnsAnObjectOfUserSummary() {
+        //arrange
+        User user = createUser(1);
+        UserSummary userSummary = new UserSummary(
+                user.getUsername(),
+                user.getName(),
+                user.getEmail(),
+                user.getGravatarHash());
+        UserPrincipal userPrincipal = new UserPrincipal(
+                currentUserId,
+                user.getName(),
+                user.getUsername(),
+                user.getEmail(),
+                user.getPassword(),
+                Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER")));
+        when(userRepository.findById(userPrincipal.getId())).thenReturn(Optional.of(user));
+
+        //act
+        UserSummary returnedUserSummary = userService.getUserInfo(userPrincipal);
+
+        //assert
+        verify(userRepository).findById(userPrincipal.getId());
+        Assert.assertEquals(userSummary.getName(), returnedUserSummary.getName());
```

Comment:
It is a good practice to override equals(), hashCode() and toString() methods in EVERY data object.
So you can add these methods to UserSummary class and check equality of entire objects

Reply:
use lombok to generate methods